### DISCUSSION TITLE
[General] Chai matcher revertedWith doesn't work

### DISCUSSION THREAD

**Discussion by novaknole** [zkSync-Community-Hub/zksync-developers#489] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

My test looks the following:

```js
const t = await deployer.loadArtifact("Test");
const contract = await deployer.deploy(t)

await expect(contract.oe()).to.be.revertedWith("oeee123")
```

My contract:

```js
contract Test  {
    function oe() public {
        require(false, "oeee123");
    }
}
```

My hardhat network:

```js
 networks: {
    hardhat: {
      throwOnTransactionFailures: true,
      throwOnCallFailures: true,
      blockGasLimit: 3000000000, 
      gasPrice: 80000000000,
      allowUnlimitedContractSize: true,
      zksync: true
    },
    zkSyncLocal: {
      url: "http://127.0.0.1:8011",
      ethNetwork: "http://127.0.0.1:8545",
      zksync: true,
      accounts: [
         "0x3d3cbc973389cb26f657686445bcc75662b415b656078503592ac8c1abb8810e"
      ],
    },
  },
```

My imports in hardhat config:
```js
import "@matterlabs/hardhat-zksync-solc";
import "@matterlabs/hardhat-zksync-deploy";
import "@matterlabs/hardhat-zksync-node";
import '@nomicfoundation/hardhat-chai-matchers';
```

My package.json:

```js
"devDependencies": {
    "@matterlabs/hardhat-zksync-deploy": "^0.8.0",
    "@matterlabs/hardhat-zksync-node": "^0.1.0",
    "@matterlabs/hardhat-zksync-solc": "^1.1.4",
    "@nomicfoundation/hardhat-chai-matchers": "^2.0.6",
    "@nomicfoundation/hardhat-ethers": "^3.0.0",
    "@typechain/ethers-v5": "^7.2.0",
    "@typechain/hardhat": "^2.3.1",
    "@types/chai": "^4.2.22",
    "@types/mocha": "^9.0.0",
    "chai": "4.3.6",
    "dotenv": "^10.0.0",
    "ethers": "^5.6.9",
    "hardhat": "^2.12.0",
    "hardhat-deploy": "0.12.4",
    "tmp-promise": "^3.0.3",
    "ts-node": "^8.1.0",
    "typechain": "^5.2.0",
    "typescript": "^4.4.4"
  },
  "dependencies": {
    "@nomicfoundation/hardhat-network-helpers": "^1.0.10",
    "@nomicfoundation/hardhat-toolbox": "^5.0.0",
    "@nomicfoundation/hardhat-verify": "^1.0.4",
    "@openzeppelin/contracts": "4.9.5",
    "@openzeppelin/contracts-upgradeable": "4.9.5",
    "jsonfile": "^6.1.0",
    "zksync-ethers": "^5.0.0"
  }
```

### What is the specific issue or error you're encountering?

Running the test fails, see below:

### Can you share the error messages or logs you're receiving, if any?

> Error: cannot estimate gas; transaction may fail or may require manual gas limit [ See: https://links.ethers.org/v5-errors-UNPREDICTABLE_GAS_LIMIT ] (error={"reason":"execution reverted: oeee123"

### Have you made any recent changes to the contract before encountering this issue?

It seems like it fails before `expect` matcher even starts to do its check.

### Are there any external libraries or contracts that your contract interacts with?

no

### Can you provide the relevant portions of your contract code where the issue is occurring?

See the reproducible example:

https://github.com/novaknole/zksync-hardhat-deploy

```js
// 1. yarn
// 2. open terminal in this project and run yarn hardhat node-zksync
// 3. yarn build --network zkSyncLocal
// 4. yarn test --network zkSyncLocal
```

For more info, I think I debugged this. [see](https://github.com/ethers-io/ethers.js/discussions/4715)

It turns out that this revertedWith won't work with ethers v5. do you agree and what could be the solution other than using v6 ?

### Have you tried to isolate the problem, and if so, what were the results?

It's isolated

### What steps have you already taken to try to resolve the issue?

Not sure what else to do.

**Comment by bxpana** [comment]
Thanks for the post, we'll get back to this as soon as possible.

**Reply by novaknole** [reply]
Note that this only happens when I use the @nomicfoundation/hardhat-chai-matchers (^1) version. If I update it to version "^2", it works, but I can't update it because updating it causes me to update all my own dependencies(ethers 5 to ethers 6) which i can't do.

**Reply by novaknole** [reply]
I also provided the reproducible repository.

**Reply by dutterbutter** [reply, marked as answer]
Currently we are only supporting `@nomicfoundation/hardhat-chai-matchers` latest version. If you require v5 usage please make use of the sunsetted `hardhat-zksync-chai-matchers` [package](https://www.npmjs.com/package/@matterlabs/hardhat-zksync-chai-matchers) instead. Specifically use version <1.0.0 which is compatible with ethers v5.
